client/events.go

```diff
@@ -46,3 +47,48 @@ func (cli *Client) Events(ctx context.Context, options types.EventsOptions) (io.
 	}
 	return serverResponse.body, nil
 }
+
+// EventsWithHandler decodes the stream of events handing them off to the appropriate
+// handlers for processing. It's up to the caller to stop the processing by canceling
+// the context. The error channel will be closed once the end of the input stream has been
+// reached. If an error is received all processing will be stopped.
+func (cli *Client) EventsWithHandler(ctx context.Context, options types.EventsOptions, handler eventtypes.EventHandler) <-chan error {
+	errChan := make(chan error)
+
+	go func() {
+		defer close(errChan)
+
+		stream, err := cli.Events(ctx, options)
+		if err != nil {
+			errChan <- err
+			return
+		}
+
+		defer stream.Close()
+
+		eventsq, errq := eventtypes.DecodeEvents(ctx, stream)
+
+		for {
+			select {
+			case <-ctx.Done():
+				// exit
+				return
+			case err := <-errq:
+				if err == io.EOF {
+					// end of input stream reached
+					return
+				}
+
+				errChan <- err
+				return
+			case event := <-eventsq:
+				processer := handler.Get(event.Action)
+				if processer != nil {
+					processer(event)
```

Comment:
Keep going back and forth here on whether or not to do this in another goroutine. Thoughts? I like _not_ because then you keep to sequential processing which can be an important thing with events.